File: frontend/src/types/fashion.ts
export interface ItemPiece {
  category: string;
  name: string;
  brand_or_label: string;
  material: string;
  color_name: string;
  color_hex: string;
  image_url: string;
  style_note: string;
  price_tag: string;
}

export interface ColorHarmony {
  scheme_name: string;
  dominant_hex: string;
  secondary_hex: string;
  accent_hex: string;
  contrast_ratio: string;
  theory_description: string;
}

export interface FacialMetrics {
  face_shape: string;
  skin_tone: string;
  undertone: string;
  jawline_angle: string;
  symmetry_score: number;
  eyewear_recommendation: string;
  grooming_advice: string;
  metal_harmony: string;
}

export interface FashionLook {
  id: string;
  look_code: string;
  title: string;
  gender: 'male' | 'female';
  culture: 'indian' | 'western' | 'fusion';
  occasion: string;
  skin_type: string;
  body_type: string;
  hair_type: string;
  face_shape: string;
  hero_image: string;
  description: string;
  color_harmony: ColorHarmony;
  pieces: ItemPiece[];
  accessories_summary: string;
  grooming_note: string;
  biometric_match_rate: number;
  tags: string[];
}

export interface MatrixQueryParams {
  gender?: string;
  culture?: string;
  skin?: string;
  body?: string;
  hair?: string;
  face_shape?: string;
  occasion?: string;
  search?: string;
  limit?: number;
  offset?: number;
}

export interface MatrixQueryResponse {
  total_matches: number;
  count: number;
  looks: FashionLook[];
  facial_geometry: FacialMetrics;
  params: MatrixQueryParams;
}

export interface SavedLookCreate {
  look_code: string;
  title: string;
  gender: string;
  culture: string;
  occasion: string;
  hero_image: string;
  description: string;
  color_harmony: ColorHarmony;
  pieces: ItemPiece[];
  accessories_summary: string;
  grooming_note: string;
  biometric_match_rate: number;
  notes?: string;
  tags: string[];
}

export interface SavedLook extends SavedLookCreate {
  id: string;
  created_at: string;
}

export interface MatrixStats {
  total_combinations: number;
  base_profiles: number;
  procedural_scale_factor: number;
  indian_looks: number;
  western_looks: number;
  male_looks: number;
  female_looks: number;
  supported_face_shapes: string[];
  supported_skin_tones: string[];
  supported_body_types: string[];
  engine_version: string;
}

File: backend/models/fashion.py
from pydantic import BaseModel, Field
from typing import List, Optional, Dict, Any
import uuid
from datetime import datetime, timezone


class ItemPiece(BaseModel):
    category: str  # top, bottom, footwear, eyewear, wristwear
    name: str
    brand_or_label: str
    material: str
    color_name: str
    color_hex: str
    image_url: str
    style_note: str
    price_tag: str


class ColorHarmony(BaseModel):
    scheme_name: str  # e.g., "Monochromatic Emerald", "Analogous Jewel", "Triadic Royal", "High-Contrast Stealth"
    dominant_hex: str
    secondary_hex: str
    accent_hex: str
    contrast_ratio: str
    theory_description: str


class FacialMetrics(BaseModel):
    face_shape: str  # Oval, Square, Heart, Diamond, Round, Oblong
    skin_tone: str  # Cool Fair, Warm Olive, Deep Rich, Golden Amber
    undertone: str  # Cool, Warm, Neutral
    jawline_angle: str  # Defined 115°, Sculpted Sharp, Soft Rounded
    symmetry_score: float  # e.g. 96.4
    eyewear_recommendation: str
    grooming_advice: str
    metal_harmony: str  # Emerald & Brushed Gold, Platinum & Cyber Steel, Rose Gold & Copper


class FashionLook(BaseModel):
    id: str = Field(default_factory=lambda: str(uuid.uuid4()))
    look_code: str  # e.g. "AS-704"
    title: str
    gender: str  # male, female
    culture: str  # indian, western, fusion
    occasion: str  # Gala, Cyber Street, Festive Wedding, Executive Soiree, Summer Resort
    skin_type: str  # fair, olive, deep
    body_type: str  # athletic, muscular, soft
    hair_type: str  # short, textured, long, curly
    face_shape: str  # oval, square, heart, diamond
    hero_image: str
    description: str
    color_harmony: ColorHarmony
    pieces: List[ItemPiece]
    accessories_summary: str
    grooming_note: str
    biometric_match_rate: float
    tags: List[str] = Field(default_factory=list)


class MatrixQueryParams(BaseModel):
    gender: str = "all"  # all, male, female
    culture: str = "all"  # all, indian, western, fusion
    skin: str = "olive"  # fair, olive, deep
    body: str = "athletic"  # athletic, muscular, soft
    hair: str = "textured"  # short, textured, long, curly
    face_shape: str = "oval"  # oval, square, heart, diamond
    occasion: Optional[str] = None
    search: Optional[str] = None
    limit: int = 24
    offset: int = 0


class SavedLookCreate(BaseModel):
    look_code: str
    title: str
    gender: str
    culture: str
    occasion: str
    hero_image: str
    description: str
    color_harmony: ColorHarmony
    pieces: List[ItemPiece]
    accessories_summary: str
    grooming_note: str
    biometric_match_rate: float
    notes: Optional[str] = ""
    tags: List[str] = Field(default_factory=list)


class SavedLook(SavedLookCreate):
    id: str = Field(default_factory=lambda: str(uuid.uuid4()))
    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))


class TelemetryEvent(BaseModel):
    id: str = Field(default_factory=lambda: str(uuid.uuid4()))
    event_type: str  # camera_scan, matrix_query, wardrobe_save, profile_change
    details: Dict[str, Any] = Field(default_factory=dict)
    timestamp: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
